cloud/sts_test.go

```diff
@@ -0,0 +1,68 @@
+package cloud
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/evergreen-ci/evergreen/db"
+	"github.com/evergreen-ci/evergreen/model/task"
+	"github.com/evergreen-ci/utility"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAssumeRole(t *testing.T) {
+	taskID := "task_id"
+	projectID := "project_id"
+	requester := "requester"
+
+	roleARN := "role_arn"
+	policy := "policy"
+	externalID := fmt.Sprintf("%s-%s", projectID, requester)
+
+	testCases := map[string]func(t *testing.T, manager STSManager, awsClientMock *awsClientMock){
+		"InvalidTask": func(t *testing.T, manager STSManager, awsClientMock *awsClientMock) {
+			_, err := manager.AssumeRole(context.Background(), taskID, AssumeRoleOptions{
```

Comment:
nit: it's almost always preferred to use a context that cancels if possible rather than a plain background context, like:
```go
ctx, cancel := context.WithCancel(context.Background())
defer cancel()
```

The reason being that if any logic that we write (or a thirdparty library uses) uses a goroutine, those goroutines will typically be allowed to run until the context cancels, but if it never cancels, then the goroutine can run in the background continuously even after the test has ended (and therefore can affect later tests). Cancelling the context at the end of the test makes it easier for tests to end any background operations that could be going on. (This reasoning about background operations also applies to the non-testing source code, which is why we try to avoid using plain `context.Background()`/`context.TODO()` as much as possible.)

For convenience, I usually just create the cancellable context in `t.Run` and pass it to the individual test cases.